src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirModel.cs

```diff
@@ -310,6 +315,36 @@ private static IEnumerable<SqlDataRecord> CreateSqlDataRecords(IEnumerable<Resou
             }
         }
 
+        private void InitializeSearchParameterStatusInformation()
+        {
+            using (var connection = new SqlConnection(_configuration.ConnectionString))
+            {
+                connection.Open();
+
+                using (SqlCommand sqlCommand = connection.CreateCommand())
+                {
+                    sqlCommand.CommandText = @"
+                        SET XACT_ABORT ON
+                        BEGIN TRANSACTION
+                        DECLARE @lastUpdated datetimeoffset(7) = SYSDATETIMEOFFSET()
+    
+                        UPDATE dbo.SearchParam
+                        SET Status = sps.Status, LastUpdated = @lastUpdated, IsPartiallySupported = sps.IsPartiallySupported
+                        FROM dbo.SearchParam INNER JOIN @searchParamStatuses as sps
+                        ON dbo.SearchParam.Uri = sps.Uri
+                        COMMIT TRANSACTION";
+
+                    IEnumerable<ResourceSearchParameterStatus> statuses = _filebasedSearchParameterStatusDataStore.GetSearchParameterStatuses().GetAwaiter().GetResult();
+
+                    SqlParameter tableValuedParameter = sqlCommand.Parameters.AddWithValue("@searchParamStatuses", CreateSqlDataRecords(statuses));
+                    tableValuedParameter.SqlDbType = SqlDbType.Structured;
+                    tableValuedParameter.TypeName = "dbo.SearchParamTableType_1";
+
+                    sqlCommand.ExecuteScalar();
```

Comment:
nit : Can we use ExecuteNonQuery instead of ExecuteScalar since we are not expecting any value to return

Reply:
Good catch, fixed in 67df5f3.